src/main/java/hudson/util/VersionNumber.java

```diff
@@ -513,16 +513,25 @@ public boolean isNewerThan(VersionNumber rhs) {
         return compareTo(rhs) > 0;
     }
 
+    /**
+     * Returns a digit (numeric component) by its position.
```

Comment:
@stephenc 

>Your version, as I understand it, will have `new VersionNumber("1.0.0.GA.2-3").digit(3) == -1`

No, we will have: `assertEquals(0, new VersionNumber("1.0.0.GA.2-3").digit(3));`